submissions/madmaxWMFU/Friends-App/js/app.js

```diff
@@ -0,0 +1,100 @@
+const allFriends = {
+    currentList: [],
+    changeList: [],
+    countFriends: 52,
+    minAge: 0,
+    maxAge: 150
+};
+
+const friendZone = document.querySelector(".friends-zone");
+const filters = document.querySelector(".filters-input");
+const sorts = document.querySelector(".search-sort-options");
+const resetFilters = document.querySelector(".reset-button");
+
+const getDataApi = async () => {
+    const response = await fetch(`https://randomuser.me/api/?results=${allFriends.countFriends}`);
+    const data = await response.json();
+    allFriends.currentList = data.results;
+    renderFriendsList(allFriends.currentList);
+}      
+
+const drawFriendsCards = (user) => {
+    let temp = `<div class="user-card shadow-profile">
+                    <div class="user-card__img">
+                        <img src="${user.picture.large}" alt="My friend">
+                    </div>
+                    <div class="user-card__cnt">
+                        <div class="profile-card__name"><span>${user.name.first}</span></div>
+                        <div class="profile-card__name"><span>${user.name.last}</span></div>
+                        <div class="profile-card__name"><span>${user.location.city}, ${user.nat}</span></div>
+                        <div class="profile-card__name"><address>Place: ${user.location.street.name}, ${user.location.street.number}</address></div>
+                    </div> 
+                    <span class="user-card__age">${user.dob.age}</span>
+                </div>`;
+    friendZone.insertAdjacentHTML("beforeEnd", temp);
+}
+
+const cleanFriendsZone = () => friendZone.innerHTML = "";
+
+const renderFriendsList = (allFriends) => allFriends.forEach(friend => drawFriendsCards(friend));
+
+const funcFilterList = {
+    filterByGender(list, {gender}) { return gender !== "all" ? list.filter((person) => person.gender === gender) : allFriends.currentList},
+    filterByAge(list, {ageTill, ageTo}) { return list.filter((person) => ageTill <= person.dob.age && person.dob.age <= ageTo)},
+    filterByName(list, {name}) { return name != "" ? list.filter((person) => new RegExp(name, 'i').test(person.name.first)) : null}
+}
+
+const getFilterList = () => {
+    allFriends.changeList = [...allFriends.currentList];
+    const searchValue = {
```

Comment:
In what case this inputs can be without value? Maybe better will be provide default value for each of them? 🤔